### DISCUSSION TITLE
What coding style is used in the project?

### DISCUSSION THREAD

**Discussion by farhaduneci** [saleor/saleor#13001] [Q&A] [answered]
Hello Saleor community!

I recently came across the Saleor project, and I must say, I am amazed by the quality and design of the codebase. The project seems to be well-structured and follows a consistent coding style, which is something I aspire to achieve in my own Django projects.

I wanted to reach out to you and inquire about the code style used in the project. I'm particularly interested in learning more about the conventions and best practices followed by the developers. 

Could you please share any information or resources that can help me understand and adopt the coding style used in Saleor?

I noticed that the Django community has a comprehensive style guide called "Django-Styleguide" (https://github.com/HackSoftware/Django-Styleguide) that provides guidelines and recommendations for writing clean and maintainable Django code. It has been immensely helpful for me in the past.

Is there a similar resource or documentation available specifically for the project? I would love to explore any guides, documentation, or style recommendations that Saleor developers have created to ensure consistency and maintainability in the codebase.

Thank you in advance for your assistance and for creating such an amazing open-source project. I look forward to your response and the opportunity to learn from the community.

Best regards,
Farhad

**Comment by maarcingebala** [comment, marked as answer]
Hey @Farhaduneci, we're glad to hear that! We pay close attention to keeping our codebase clean to make it easier to maintain and add new changes. You can find our best practices in the [Contributing Guide](https://docs.saleor.io/docs/3.x/developer/community/contributing) in the documentation. Feel free to use it as inspiration for your own projects. In my opinion, the following practices provide the most benefits:

- Using linters and code formatting tools like `black`.
- Running `pre-commit` checks locally before pushing changes to the repository.
- Having a well-configured CI that runs all checks and test suites.
- Dividing files early into smaller ones to keep them manageable.

Hope it helps, thanks!

**Reply by farhaduneci** [reply]
Thank you for your response! I appreciate the insight, the contributing guide was what I've been exactly searching for but missed it.
